src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -116,6 +121,15 @@ void GcsNodeManager::NodeFailureDetector::SendBatchedHeartbeat() {
       }
     }
 
+    // Update placement group load to heartbeat batch.
+    // This is updated only one per second.
```

Comment:
Wouldn't it be simpler to just report placement groups with each heartbeat? We can avoid adding too much overhead by adding a max_num_placement_groups config to restrict the max number of groups to report at once.

Reply:
> restrict the max number of groups to report at once.

I will definitely implement this. But I don't really understand what makes it simpler if I report this to every heartbeat. Is it because autoscaler can have a better view of the placement group load in real-time? 

I feel like it is common to have many large placement groups (maybe >= 100 bytes) in large and multi tenant clusters, and doing this calculation every 100ms seems to be a bit expensive.

Reply:
I don't have a strong opinion, and it definitely could be premature optimization though. 

So, if you think it is fine, I will just report for every heartbeat, but I'd like to understand what simplicity benefits we can have.